tools/PI/DevHome.PI/Helpers/ExternalTool.cs

```diff
@@ -115,8 +96,38 @@ private async void GetIcons()
             _log.Error(ex, "Failed to get tool image");
         }
     }
+
+    public override IconElement GetIcon()
+    {
+        return new ImageIcon
+        {
+            Source = ToolIconSource,
+        };
+    }
+
+    internal async override void InvokeTool(Window? parentWindow, int? targetProcessId, HWND hWnd)
+    {
+        try
+        {
+            var process = await InvokeToolInternal(targetProcessId, hWnd);
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, "Tool launched failed");
+
+            var builder = new StringBuilder();
+            builder.AppendLine(ex.Message);
+            if (ex.InnerException is not null)

```

Comment:
[Exception.ToString()](https://learn.microsoft.com/en-us/dotnet/api/system.exception.tostring?view=net-8.0#remarks) prints the inner exception as well.

Reply:
I thought this comment sounded familiar. This is code being moved from another file, and this was Andrew's response to this feedback:

"You're correct it does, but it also includes a callstack, which in this instance isn't useful - this scenario just needs a simple message stating the root cause."